pymc4/random_variables/continuous.py

```diff
@@ -42,6 +49,165 @@ def _base_dist(self, *args, **kwargs):
+        """
+        # TODO: if skew=None, make a 0 skew.
+        # is corr foreced to equal 1 when univariate?
+        parameters = dict(locals())
+        #with ops.name_scope(name, values=[loc, scale]) as name:
+        #  with ops.control_dependencies([check_ops.assert_positive(scale)] if
+        #                                validate_args else []):
+        #   self._corr = array_ops.identity(corr, name="correlation")
+        #    self._skew = array_ops.identity(skew, name="skew")
+        #    check_ops.assert_same_float_dtype([self._loc, self._scale])
+        
+        
+        self._corr = array_ops.identity(corr, name="correlation")
+        self._skew = array_ops.identity(skew, name="skew")
+        # assume matrix
+        if self._corr.ndim == 2:
+            self._corr = expand_dims(self._corr, 0)
+        # assume vec 
+        if self._skew.ndim == 1:
+            self._skew = expand_dims(self._skew, 0)
+        
+                
+        super(StdSkewNormal, self).__init__(
+            dtype=self._corr.dtype,
+            reparameterization_type=distribution.FULLY_REPARAMETERIZED, # ?
+            validate_args=validate_args,
+            allow_nan_stats=allow_nan_stats,
+            parameters=parameters,
+            graph_parents=[self._corr, self._skew],
+            name=name)
+    
+    @property
+    def corr(self):
+        """correlation matrix"""
+        return self._corr
+    
+    @property
+    def Omega(self):
+        shape = concat([self.batch_shape_tensor(), array_ops.shape(self.corr)[-2:]], 0)
+        return broadcast_to(self.corr, shape)
+    
+    # why doesn't TFP Distribution do this?
+    @property
+    def skew(self):
+        """skew vector"""
+        return self._skew
+    
+    @property
+    def alpha(self):
+        alpha = expand_dims(self.skew, -1) # makes a (col., n x 1) vector 
+        shape = concat([self.batch_shape_tensor(), array_ops.shape(alpha)[-2:]], 0)
+        return broadcast_to(alpha, shape)
+    
+    def _mean(self):
+        return (2./pi)**(.5) * self.get_delta()
+    
+    def _variance(self):
+        mu = self.mean()
+        return self.Omega - matmul(mu, mu, transpose_b=True)
+    
+    #@property # is property a good idea here, since there are computations involved?
+    #def delta(self):
+     #   return get_delta if not calculated
+    
+    def get_delta(self):
+        Omegaalpha = self.Omega @ self.alpha
+        return Omegaalpha @ (1 + matmul(self.alpha, Omegaalpha, transpose_a=True))**(-.5)
+    
+    #@property
+    def get_Omega_star(self):
+        """correlation matrix of underlying normal distribution"""
+        delta = self.get_delta()
+        Omega = self.Omega
+        one_shp = array_ops.shape(delta)
+        # one_shp[-2] = 1 doesn't work
+        # https://github.com/tensorflow/tensorflow/issues/14132
+        # so instead, ...
+        one_shp = concat([one_shp[:-2] , [1,1] ], 0)
+        one = ones(one_shp)
```

Comment:
you could do array_ops.shape(delta[..., :1, :1])

also generally outside TF we recommend using API references like tf.shape(..) instead of array_ops.shape(..)